virtcontainers/device/drivers/vfio.go

```diff
@@ -187,8 +187,18 @@ func (device *VFIODevice) Load(ds persistapi.DeviceState) {
 
 // It should implement GetAttachCount() and DeviceID() as api.Device implementation
 // here it shares function from *GenericDevice so we don't need duplicate codes
+
 func getVFIODetails(deviceFileName, iommuDevicesPath string) (deviceBDF, deviceSysfsDev string, vfioDeviceType config.VFIODeviceType, err error) {
-	vfioDeviceType = GetVFIODeviceType(deviceFileName)
+	tokens := strings.Split(deviceFileName, ":")
```

Comment:
It might be worth putting this logic into a new function that can be unit-tested with various numbers of colons and dashes in `deviceFileName`.

```go
func getVFIODeviceType(deviceFileName string) string
```